src/MOLFiniteDifference/MOL_discretization.jl

```diff
@@ -132,18 +139,48 @@ function SciMLBase.symbolic_discretize(pdesys::ModelingToolkit.PDESystem,discret
         valrules = vcat([depvars[k] => depvarsdisc[k][II] for k in 1:length(depvars)],
                         [nottime[j] => space[j][II[j]] for j in 1:nspace])
     
-        # TODO: Use rule matching for nonlinear Laplacian
-    
         # TODO: upwind rules needs interpolation into `@rule`
-        #forward_weights(i,j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, [space[j][i[j]],space[j][i[j]+1]])
-        #reverse_weights(i,j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, [space[j][i[j]-1],space[j][i[j]]])
-        #upwinding_rules = [@rule(*(~~a,(Differential(nottime[j]))(u),~~b) => IfElse.ifelse(*(~~a..., ~~b...,)>0,
-        #                        *(~~a..., ~~b..., dot(reverse_weights(i,j),depvarsdisc[k][central_neighbor_idxs(i,j)[1:2]])),
-        #                        *(~~a..., ~~b..., dot(forward_weights(i,j),depvarsdisc[k][central_neighbor_idxs(i,j)[2:3]]))))
-        #                        for j in 1:nspace, k in 1:length(depvars)]
-    
-        rules = vcat(vec(central_deriv_rules),valrules)
+        forward_weights(i,j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, [space[j][i[j]],space[j][i[j]+1]])
+        reverse_weights(i,j) = DiffEqOperators.calculate_weights(discretization.upwind_order, 0.0, [space[j][i[j]-1],space[j][i[j]]])
+        # upwinding_rules = [@rule(*(~~a,$(Differential(nottime[j]))(u),~~b) => IfElse.ifelse(*(~~a..., ~~b...,)>0,
+        #                         *(~~a..., ~~b..., dot(reverse_weights(i,j),depvars[k][central_neighbor_idxs(i,j)[1:2]])),
+        #                         *(~~a..., ~~b..., dot(forward_weights(i,j),depvars[k][central_neighbor_idxs(i,j)[2:3]]))))
+        #                         for j in 1:nspace, k in 1:length(pdesys.depvars)]
+
+        ## Discretization of non-linear laplacian. 
+        # d/dx( a du/dx ) ~ (a(x+1/2) * (u[i+1] - u[i]) - a(x-1/2) * (u[i] - u[i-1]) / dx^2
+        b1(i, j, k) = dot(reverse_weights(i, j), depvarsdisc[k][central_neighbor_idxs(i, j)[1:2]]) / dxs[j]
+        b2(i, j, k) = dot(forward_weights(i, j), depvarsdisc[k][central_neighbor_idxs(i, j)[2:3]]) / dxs[j]
+        # TODO: improve interpolation of g(x) = u(x) for calculating u(x+-dx/2)
+        g(i, j, k, l) = sum([depvarsdisc[k][central_neighbor_idxs(i, j)][s] for s in (l == 1 ? [2,3] : [1,2])]) / 2.
+        # iv_mid returns middle space values. E.g. x(i-1/2) or y(i+1/2).
+        iv_mid(i, j, l) = (space[j][i[j]] + space[j][i[j]+l]) / 2.0 
```

Comment:
note harmonic mean might be more accurate http://web.cecs.pdx.edu/~gerry/class/ME448/lecture/pdf/CVFDdiffusion2D.pdf

Reply:
Thanks for the note. I added to my [draft](https://docs.google.com/document/d/1qbWokDYXuO5GV3ilI8R4fWWnvkBkrpQd3H6d7l9DrC4/edit).